cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4CareGapsProcessor.java

```diff
@@ -0,0 +1,182 @@
+    private final R4MeasureServiceUtils r4MeasureServiceUtils;
+    private final R4CareGapsBundleBuilder r4CareGapsBundleBuilder;
+
+    public R4CareGapsProcessor(
+            CareGapsProperties careGapsProperties,
+            Repository repository,
+            MeasureEvaluationOptions measureEvaluationOptions,
+            String serverBase) {
+        this.repository = repository;
+        this.careGapsProperties = careGapsProperties;
+        this.measureEvaluationOptions = measureEvaluationOptions;
+        this.serverBase = serverBase;
+
+        r4MeasureServiceUtils = new R4MeasureServiceUtils(repository);
+        r4CareGapsBundleBuilder = new R4CareGapsBundleBuilder(
+                careGapsProperties, repository, measureEvaluationOptions, serverBase, configuredResources);
+    }
+
+    public Parameters getCareGapsReport(
+            IPrimitiveType<Date> periodStart,
+            IPrimitiveType<Date> periodEnd,
+            String subject,
+            List<String> statuses,
+            List<IdType> measureIds,
+            List<String> measureIdentifiers,
+            List<CanonicalType> measureUrls) {
+
+        // validate and set required configuration resources for care-gaps
+        checkConfigurationReferences();
+
+        // Collect Measures to Evaluate
+        List<Measure> measures =
+                r4MeasureServiceUtils.getMeasures(measureIds, measureIdentifiers, canonicalToString(measureUrls));
+        List<IdType> collectedMeasureIds =
+                measures.stream().map(Resource::getIdElement).collect(Collectors.toList());
+
+        // validate required parameter values
+        checkValidStatusCode(statuses);
+        measureCompatibilityCheck(measures);
+
+        // Subject Population for Report
+        List<String> subjects = getSubjects(subject);
+
+        // Build Results
+        Parameters result = initializeResult();
+
+        // Build Patient Bundles
+
+        List<Parameters.ParametersParameterComponent> components = r4CareGapsBundleBuilder.makePatientBundles(
+                periodStart, periodEnd, subjects, statuses, collectedMeasureIds);
+
+        // Return Results with Bundles
+        return result.setParameter(components);
+    }
+
+    protected List<String> getSubjects(String subject) {
+        R4RepositorySubjectProvider subjectProvider = new R4RepositorySubjectProvider();
+        var subjects = subjectProvider.getSubjects(repository, null, subject).collect(Collectors.toList());
+        if (!subjects.isEmpty()) {
+            ourLog.info(String.format("care-gaps report requested for: %s subjects.", subjects.size()));
+        } else {
+            ourLog.info("care-gaps report requested for: 0 subjects.");
+        }
+        return subjects;
+    }
+
+    private void addConfiguredResource(String id, String key) {
+        // read resource from repository
+        Resource resource = repository.read(Organization.class, new IdType(id));
```

Comment:
Suggestion:  use the IdType constructor with the resource type and pass in "Organizatoin"